keps/sig-network/3070-reserved-service-ip-range/README.md

```diff
@@ -0,0 +1,444 @@
+
+The proposal is to implement a new strategy on the Service IP allocation bitmap that logically subdivide the range
+in two bands based on the following formula `min(max($min, cidrSize/$step), $max)`, described as ~never less than $min or more than $max, with a graduated step function between them~, with $min = 16, $max = 256 and $step = 16.
+
+- lower band, used preferably for static ip assignment.
+- upper band, used preferably for dynamic IP assignment.
+
+Dynamically IP assignment will use the upper band by default, once this has been exhausted it will use the
+lower range. This will allow users to define static allocations on the lower band of the predefined
+range with a low risk of collision, keeping the implementation backwards compatible.
+
+Example 1 Large Subnet:
+
+- Service IP CIDR: 192.168.0.0/16
+- Range Size: 2^16 - 2 = 65534
+- Band Offset: min(max(16,65536/16),256) = min(4096,256) = 256
+- Static band start: 192.168.0.1
+- Static band ends: 192.168.1.1
+
+           ┌─────────────┬─────────────────────────────────────────────┐
+           │   static    │                    dynamic                  │
+           └─────────────┴─────────────────────────────────────────────┘
+
+           ◄────────────► ◄────────────────────────────────────────────►
+    192.168.0.1     192.168.1.1                                 192.168.255.254
+
+Example 2:
+
+- Service IP CIDR: 192.168.0.0/22
+- Range Size: 2^10 - 2 = 1022
+- Band Offset: min(max(16,1024/16),256) = min(max(16,64),256) = 64
+- Static band start: 192.168.0.1
+- Static band ends: 192.168.0.64
+
+           ┌─────────────┬─────────────────────────────────────────────┐
+           │   static    │                    dynamic                  │
+           └─────────────┴─────────────────────────────────────────────┘
+
+           ◄────────────► ◄────────────────────────────────────────────►
+    192.168.0.1     192.168.0.64                                 192.168.3.254
+
+Example 3 Small Subnet:
+
+- Service IP CIDR: 192.168.0.0/26
+- Range Size: 2^6 - 2 = 62
+- Band Offset: min(max(16,64/16),256) = min(max(16,4),256) = 16
+- Static band start: 192.168.0.1
+- Static band ends: 192.168.0.16
+
+           ┌─────────────┬─────────────────────────────────────────────┐
+           │   static    │                    dynamic                  │
+           └─────────────┴─────────────────────────────────────────────┘
+
+           ◄────────────► ◄────────────────────────────────────────────►
+    192.168.0.1     192.168.0.16                                 192.168.0.62
+
+
+```go
+// randomScanReservedStrategy choose a random address from the provided big.Int and then scans
+// forward looking for the next available address. The big.Int range is subdivided so it will try
+// to allocate first from the reserved block of addresses (it will wrap the reserved subrange if necessary).
+// If there is no free address it will try to allocate one from the reserved block too.
```

Comment:
"to allocate first from the **un**reserved block", right? (You're referring to both ranges as "the reserved block" here.)

and "(it will wrap around back to the start of the unreserved subrange if necessary)"

Reply:
I will word it as reserved for static and reserved for dynamic allocation, it sounds less ambiguous